libfwupdplugin/fu-i2c-device.c

```diff
@@ -145,11 +138,11 @@ fu_i2c_device_probe(FuDevice *device, GError **error)
 					      NULL))
 		return FALSE;
 
-#ifdef HAVE_GUDEV
 	/* get bus number out of sysfs path */
-	udev_parent = g_udev_device_get_parent(udev_device);
+	udev_parent =
+	    fu_udev_device_get_parent_with_subsystem(FU_UDEV_DEVICE(device), NULL, NULL, NULL);
```

Comment:
@superm1 I suspect this one should be filled in with subsystem+driver too. Will fire up some real hardware tomorrow unless you have anything handy.

Reply:
I think my Kria board will use this (amd-kria derives from i2c)

Reply:
https://github.com/user-attachments/files/16170762/udev.txt has the udev output so you can see the parents

Reply:
`/devices/platform/axi/ff030000.i2c/i2c-1/1-0050` is the interesting device

Reply:
so parent of that i2c is 'platform'